File: lib/backend/app/server/models/inside_workout_model.py
from beanie import Document
from pydantic import EmailStr
from typing import List, Optional
from datetime import datetime
class InsideWorkout(Document):
    class Settings:
        name = "InsideWorkout"
    date : str  # Date of the day
    firebase_uid : str
    description : str  # Description of the workout
    thoughts: str  # Thoughts on the workout
    completed: bool
    workoutType: str

    

File: lib/features/tracking/data/models/inside_workout_model.dart
class InsideWorkout {
  String? date;
  String? firebaseUid;
  String? description;
  String? thoughts;
  bool? completed = false;
  String? workoutType;

  InsideWorkout({
    this.date,
    this.firebaseUid,
    this.description,
    this.thoughts,
    this.completed,
    this.workoutType,
  });

  factory InsideWorkout.fromJson(Map<String, dynamic> json) {
    return InsideWorkout(
      date: json['date'],
      firebaseUid: json['firebase_uid'],
      description: json['description'],
      thoughts: json['thoughts'],
      completed: json['completed'],
      workoutType: json['workoutType'],
    );
  }
  // to json
  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'firebase_uid': firebaseUid,
      'description': description,
      'thoughts': thoughts,
      'completed': completed,
      'workoutType': workoutType,
    };
  }
}
